### DISCUSSION TITLE
Can I use a hookwrapper to append failures?

### DISCUSSION THREAD

**Discussion by amitw-di** [pytest-dev/pytest#11405] [Q&A] [answered]
Hi, 
I'm trying to implement a system-wide check that should run after each test and fail it if some conditions are met. 
I have managed to do that using code similar to the following:
```
@pytest.hookimpl(hookwrapper=True)  
def pytest_runtest_call(item):
    outcome = yield
    post_check():
       
def post_check():
    pytest.fail("Normally I'd put here some real logic")
```

However, how do I deal with tests that have already failed? I still want to execute them, but I don't want to override the test fail.
Same thing in case the test is broken (exception that is not an assertion)
Is there a way to report multiple fails on this test? 

So far the best I could think of is to fail the test only if it hadn't failed yet, with the following manner:
```
@pytest.hookimpl(hookwrapper=True)  
def pytest_runtest_call(item):
    outcome = yield
    try:
        post_check():
    except BaseException: 
        if outcome.excinfo is None:
            logger.exception("Failed in post_check")
            raise
```

is there a better way? something that will make the error more visible?

**Comment by RonnyPfannschmidt** [comment, marked as answer]
a possible way to do this is to have a fixture for it that fails in teardown
another way is to create a failure report or extend one in the hook to make test reports

**Reply by amitw-di** [reply]
Thanks for the reply. 

Moving it out of a fixture was my starting motivation, so putting it back there will not help in my case :) 
I'm not sure I know how to approach extending a failure report, could you point me to an example to understand how to start with this?

**Reply by RonnyPfannschmidt** [reply]
In not aware of a preexisting example, typically people solve this via teardown
